### PR TITLE
Story32

### PR DESCRIPTION
**Pull request by pleonar1**
Finished

### CODE REVIEW

**Comment by pleonar1** [comment]
There are merge conflicts but I just looked at them and they are harmless.  I can't fix them until I am about to merge though

File: app/models/item.rb [hunk 1]
```diff
@@ -3,10 +3,13 @@ class Item < ApplicationRecord
   has_many :invoice_items
   has_many :invoices, through: :invoice_items
 
+  enum status: [:Disabled, :Enabled]
```
Comment (jaw772) [inline app/models/item.rb:6]: does it matter if these values are capitalized?

File: app/controllers/merchant_items_controller.rb [hunk 2]
```diff
@@ -17,18 +17,30 @@ def edit
   def update
     merchant = Merchant.find(params[:merchant_id])
     item = Item.find(params[:item_id])
-    item.update(item_params)
-    # flash[:notice] = "Item Successfully Updated"
-    # binding.pry
-    # if item.save
-    redirect_to "/merchants/#{merchant.id}/items/#{item.id}",  notice: "Item Successfully Updated"
-    # else
-    # end
```
Comment (jaw772) [inline app/controllers/merchant_items_controller.rb:26]: good job on cleaning this up

File: app/views/merchant_items/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<h1>New Item for <%= @merchant.name %></h1>
+<div id=create_item>
+  <%= form_with url: "/merchants/#{@merchant.id}/items", method: :post, local: true do |form| %>
```
Comment (jaw772) [inline app/views/merchant_items/new.html.erb:3]: We will have to change this in the refactor to form_with model style

**Review by jaw772 — APPROVED** [review]

File: app/models/item.rb (hunk 1 above)
Comment (LelandCurtis) [reply app/models/item.rb:6]: this post suggest camel case naming convention for enums. so lowercase. https://stackoverflow.com/questions/40152468/enums-in-rails-uppercase-or-lowercase

File: config/routes.rb [hunk 4]
```diff
@@ -1,10 +1,15 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  
   get '/merchants/:id/items', to: 'merchant_items#index'
```
Comment (LelandCurtis) [inline config/routes.rb:5]: looks like an opportunity to refactor with advanced routing.

File: db/migrate/20220106181004_add_status_to_items.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddStatusToItems < ActiveRecord::Migration[5.2]
```
Comment (LelandCurtis) [inline db/migrate/20220106181004_add_status_to_items.rb:1]: didn't you add this migration already in a different PR? Might be worth double checking that we don't have two columns added.

File: spec/factories/items.rb [hunk 6]
```diff
@@ -3,6 +3,7 @@
     sequence(:name) { |n| "Default Item Name #{n}" }
     description { "Default Description" }
     sequence(:unit_price) { |n| 10000 + (n * 1000) }
+    status { 0 }  #check with Leland on this
```
Comment (LelandCurtis) [inline spec/factories/items.rb:6]: this makes me happy! perfect.

File: spec/features/merchants/items/create_spec.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+    within('#create_item') do
+      fill_in "Name:", with: "Paul's Item"
+      fill_in "Description", with: "An item from Paul"
+      fill_in "Unit Price:", with: 1000
+      click_button "Submit"
+    end
+
+    expect(current_path).to eq("/merchants/#{merchant1.id}/items")
+    save_and_open_page
```
Comment (LelandCurtis) [inline spec/features/merchants/items/create_spec.rb:17]: please delete the save_and_open_page. haha these are so hard to find.

**Review by LelandCurtis — COMMENTED** [review]
Can you double check the migrations? I want to make sure the two different PRs you submitted aren't somehow duplicating the migration. I doubt it is, but that would suck, so worth double checking.

**Review by EMielke76 — APPROVED** [review]

**Pull request merged**